### PR TITLE
CAC Feature

### PR DESCRIPTION
**Pull request by marklise**
This brings in all the public CAC Feature components.

Nothing to be removed/disabled after approval as it's only visible once an administrator enables the feature (which will be disabled until it lands and passes test).

### CODE REVIEW

File: src/app/project/cac/become-a-member.component.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import { Component, Input, OnInit } from '@angular/core';
+// import { HttpEventType } from '@angular/common/http';
```
Comment (mtCarto) [inline src/app/project/cac/become-a-member.component.ts]: Is something planned for this import? If not, please remove

Comment (marklise) [reply src/app/project/cac/become-a-member.component.ts]: Removed.

File: src/app/project/cac/become-a-member.component.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+  ngOnInit() {
+    this.comment = new Comment();
+  }
+
+  private p1_next() {
+    this.currentPage++;
+  }
+
+  // TODO: Have some null checks in here from the front end
```
Comment (KitArmstrong) [inline src/app/project/cac/become-a-member.component.ts]: Do null checks still need to be added or is this TODO complete?

File: src/app/project/cac/become-a-member.component.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    this.projectService.cacSignUp(this.project, signUpObject)
+    .toPromise()
+    .then((res: any) => {
+      console.log('Success:', res);
+      this.submitting = false;
+      this.currentPage++;
+    })
+    .catch(error => {
+      console.log('error', error);
+      alert('Uh-oh, error submitting informaiton');
```
Comment (KitArmstrong) [inline src/app/project/cac/become-a-member.component.ts]: 'information'

File: src/app/project/comments/add-comment/add-comment.component.ts [hunk 4]
```diff
@@ -87,16 +102,76 @@ export class AddCommentComponent implements OnInit {
     }
   }
 
+  private learnMore() {
+    this.currentPage = 2;
+  }
+
   private p1_next() {
-    this.currentPage++;
+    // TODO: Check if they already submitted something, if so, skip
```
Comment (KitArmstrong) [inline src/app/project/comments/add-comment/add-comment.component.ts]: Looks like this TODO can be removed.

File: src/app/project/comments/add-comment/add-comment.component.ts [hunk 5]
```diff
@@ -87,16 +102,76 @@ export class AddCommentComponent implements OnInit {
...
+    .toPromise()
+    .then((res: any) => {
+      console.log('Success:', res);
+      this.submitting = false;
+      this.submittedCAC = true;
+      this.currentPage++;
+    })
+    .catch(error => {
+      console.log('error', error);
+      alert('Uh-oh, error submitting informaiton');
```
Comment (KitArmstrong) [inline src/app/project/comments/add-comment/add-comment.component.ts]: 'information'

File: src/app/project/project.component.ts [hunk 6]
```diff
@@ -316,6 +317,23 @@ export class ProjectComponent implements OnInit, OnDestroy, AfterViewInit {
     });
   }
 
+  public learnMore() {
+    this.ngbModal = this.modalService.open(BecomeAMemberComponent, { backdrop: 'static', size: 'lg' });
+    // set input parameter
+    (<BecomeAMemberComponent>this.ngbModal.componentInstance).project = this.project;
+    // check result
+    this.ngbModal.result.then(
```
Comment (KitArmstrong) [inline src/app/project/project.component.ts]: It doesn't appear this the result is being handled in a way other than a log? Should something else be happening in this function?

Comment (marklise) [reply src/app/project/project.component.ts]: The modal encapsulates and handles all the code required to submit a user to a CAC on a project, so the logging really is the only thing going on.  If the user cancels, it does nothing, and if it submits successfully,

File: src/app/project/comments/add-comment/add-comment.component.ts (hunk 5 above)
Comment (marklise) [reply src/app/project/comments/add-comment/add-comment.component.ts]: Fixed in 974030a.

File: src/app/project/comments/add-comment/add-comment.component.ts (hunk 4 above)
Comment (marklise) [reply src/app/project/comments/add-comment/add-comment.component.ts]: Fixed in 974030a.

File: src/app/project/cac/become-a-member.component.ts (hunk 3 above)
Comment (marklise) [reply src/app/project/cac/become-a-member.component.ts]: Fixed in 974030a.

File: src/app/project/cac/become-a-member.component.ts (hunk 2 above)
Comment (marklise) [reply src/app/project/cac/become-a-member.component.ts]: Removing comment, fixed in 974030a.

**Pull request merged**